File: 414-third-maximum-number/third-maximum-number.java
class Solution 
{
    public int thirdMax(int[] nums) 
    {
       SortedSet<Integer> set = new TreeSet<>();

       for(int i : nums) set.add(i);

       if(set.size() == 3) return set.first();

       else if(set.size() < 3) return set.last();

       else
       {
            List<Integer> list = new ArrayList<>(set);
            return list.get(list.size()-3);
       }
    }
}

File: 414-third-maximum-number/third-maximum-number.cs
public class Solution 
{
    public int ThirdMax(int[] nums) 
    {
        SortedSet<int> set = new SortedSet<int>();

        foreach(int num in nums)
        {
            set.Add(num);
        }

        if (set.Count == 3) 
            return set.Min;

        else if (set.Count < 3) 
            return set.Max;

        else
        {
            List<int> list = new List<int>(set);
            return list[list.Count - 3];
        }
    }
}